cpp/include/cugraph_c/algorithms.h

```diff
@@ -194,20 +197,82 @@ void cugraph_bfs_result_free(cugraph_bfs_result_t* result);
  * @param depth_limit Sets the maximum number of breadth-first search iterations. Any vertices
  * farther than @p depth_limit hops from @p source_vertex will be marked as unreachable.
  * @param do_expensive_check A flag to run expensive checks for input arguments (if set to `true`).
- * @param [out] result       Opaque pointer to pagerank results
+ * @param [out] result       Opaque pointer to bfs results
  * @param [out] error        Pointer to an error object storing details of any error.  Will
  *                           be populated if error code is not CUGRAPH_SUCCESS
  * @return error code
  */
-cugraph_error_code_t cugraph_bfs(const cugraph_resource_handle_t* handle,
-                                 cugraph_graph_t* graph,
-                                 cugraph_type_erased_device_array_t* sources,
-                                 bool_t direction_optimizing,
-                                 size_t depth_limit,
-                                 bool_t do_expensive_check,
-                                 bool_t compute_predecessors,
-                                 cugraph_bfs_result_t** result,
-                                 cugraph_error_t** error);
+cugraph_error_code_t cugraph_bfs(
+  const cugraph_resource_handle_t* handle,
+  cugraph_graph_t* graph,
+  // FIXME:  Make this const, copy it if I need to temporarily modify internally
+  cugraph_type_erased_device_array_t* sources,
+  bool_t direction_optimizing,
+  size_t depth_limit,
+  bool_t do_expensive_check,
+  bool_t compute_predecessors,
+  cugraph_bfs_result_t** result,
+  cugraph_error_t** error);
+
+/**
+ * @brief     Opaque extract_bfs_paths result type
+ */
+typedef struct {
+  int align_;
+} cugraph_extract_bfs_paths_result_t;
```

Comment:
Reminder for either a `FIXME` or an update in this PR to address [this](https://github.com/rapidsai/cugraph/pull/1955#issuecomment-974584672).